### PR TITLE
Reduce duplicate code for TextChange overlaps

### PR DESCRIPTION
**Pull request by andy-ms**
These functions all compute the same thing in slightly different ways.

### CODE REVIEW

File: src/compiler/utilities.ts [hunk 1]
```diff
@@ -3793,12 +3786,9 @@ namespace ts {
     }
 
     export function textSpanIntersection(span1: TextSpan, span2: TextSpan) {
-        const intersectStart = Math.max(span1.start, span2.start);
-        const intersectEnd = Math.min(textSpanEnd(span1), textSpanEnd(span2));
-        if (intersectStart <= intersectEnd) {
-            return createTextSpanFromBounds(intersectStart, intersectEnd);
-        }
-        return undefined;
```
Comment (aastikta28) [inline src/compiler/utilities.ts]: You haven't done any logic changes here. From what I can see, renamed the variables and re written the logic using ternary operator. Is there anything am missing here?

File: src/compiler/utilities.ts [hunk 2]
```diff
@@ -3759,27 +3759,20 @@ namespace ts {
...
 
     export function textSpanOverlap(span1: TextSpan, span2: TextSpan) {
-        const overlapStart = Math.max(span1.start, span2.start);
-        const overlapEnd = Math.min(textSpanEnd(span1), textSpanEnd(span2));
-        if (overlapStart < overlapEnd) {
-            return createTextSpanFromBounds(overlapStart, overlapEnd);
-        }
-        return undefined;
+        const overlap = textSpanIntersection(span1, span2);
+        return overlap && overlap.length === 0 ? undefined : overlap;
```
Comment (aastikta28) [inline src/compiler/utilities.ts]: here maybe consider doing return textSpanIntersection(span1, span2) as it returns undefined if no value.

Comment (andy-ms) [reply src/compiler/utilities.ts]: `textSpanIntersection` doesn't return undefined if there's a 0-size intersection (the ranges are right up against each other), but this function does. So they're not exactly the same.

File: src/compiler/utilities.ts (hunk 1 above)
Comment (andy-ms) [reply src/compiler/utilities.ts]: Yeah, this PR isn't meant to change the return values of anything.

**Pull request merged**